src/Core/ServiceWrapper/Program.cs

```diff
@@ -658,6 +671,39 @@ private static string ReadPassword()
             }
         }
 
+        private static ConfigType GetConfigType(out string baseName, out DirectoryInfo d)
```

Comment:
I don't think we need an enum here. Simply construct the service descriptor and return it.